### PR TITLE
Update to Foundation 6

### LINKED ISSUE #121: Support for Foundation 6
http://foundation.zurb.com/apps/docs/#!/

This will need to be verified since I have not tried it yet.

### LINKED ISSUE #131: alias_method_chain is deprecated
DEPRECATION WARNING: alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super. (called from <module:FormHelper> at /Users/dsandstrom/Projects/einstein_video5/vendor/bundle/ruby/2.3.0/bundler/gems/foundation_rails_helper-646bd8477bf6/lib/foundation_rails_helper/action_view_extension.rb:19)
DEPRECATION WARNING: alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super. (called from <module:FormHelper> at /Users/dsandstrom/Projects/einstein_video5/vendor/bundle/ruby/2.3.0/bundler/gems/foundation_rails_helper-646bd8477bf6/lib/foundation_rails_helper/action_view_extension.rb:20)

### PR DESCRIPTION
**Pull request by dsandstrom**
Foundation 6 is out so we need to make this gem compatible.

Fixes #121 

#### TODO
- [x] Add tag for current master and push to rubygems
- [x] Fix #131 
- [x] Update Readme with new classes
- [x] Make code Rubocop compliant

#### Discuss
* Should we use v2.0.0 for Foundation 6 version?

### CODE REVIEW

**Comment by dsandstrom** [comment]
@dgmstuart I added a PR for the merge.  Feel free to provide feedback or add TODOs.

**Comment by dgmstuart** [comment]
Yes, Foundation 6 should definitely be a major version bump, since it will involve breaking changes 

I feel like we should continue the convention of only supporting the latest Rails and Foundation versions, and having unsupported branches for older versions. Migrating an existing project to 6 might be painful because [there isn't a migration guide yet](http://foundation.zurb.com/forum/posts/36152-foundation-6-upgrade-docs?page=2), but I feel like the trade-off is worth it.

I see the Rails 5 stuff is rolled into here as well: I guess it's just coincidence that foundation 6 and rails 5 are happening at (very) roughly the same time, but I guess it would make sense to have one supported branch (Foundation 6/Rails 5) and one new legacy branch (Foundation 5/Rails 4.2) - how do you feel about that?

On the other hand, is rolling in Rails 5 raising the barrier to entry of an upgrade by saying "you can't use Foundation 6 unless you also upgrade Rails"?

Am I right in thinking that there's no reliable way for the gem to complain if it's not using the expected version of Foundation?

**Comment by dsandstrom** [comment]
Thanks for the feedback.

I reverted anything that had to do with Rails 5 before merging into this *foundation-6* branch.  However, if another PR gets submitted (that we approve), I will merge it.  Yet, I'd rather wait until we have an official release.

I think we can support Rails ~>4.1 and 5, for the time being.  I'm not currently using 5 on any projects, so I can't say for certain.  If it comes down to it, we can add a F6/Rails4 branch, but hopefully it's not needed.

I was looking for a migration guide, also.  Luckily, I have a project that is still in development so I could test out F6 on it.  The upgrade wasn't too bad.

**Comment by dgmstuart** [comment]
Rebased and squashed one of the 'fix' commits. 

Most of the commits went away, since they were either back-and-forth changes to the gemspec, or were already on master.

**Comment by dgmstuart** [comment]
Regarding ruby versions, it's unclear what the `foundation-rails` gem itself supports: 
- travis only tests against ruby 2.2, 
- there's a ruby-version file (used in development?) which is set to 2.1.2.

I've raised an issue asking what they support: https://github.com/zurb/foundation-rails/issues/199

File: lib/foundation_rails_helper/flash_helper.rb [hunk 1]
```diff
@@ -31,15 +34,25 @@ def display_flash_messages(key_matching = {})
   private
 
     def alert_box(value, alert_class)
-      content_tag :div, :data => { :alert => "" }, :class => "alert-box #{alert_class}" do
+      content_tag(
+        :div,
+        :class => "callout #{alert_class}",
+        :data => { closable: '' }
```
Comment (dgmstuart) [inline lib/foundation_rails_helper/flash_helper.rb]: Do we definitely want this to be on by default?

**Comment by dsandstrom** [comment]
I'll look into cherry picking your #142 commits into this branch.

**Comment by dsandstrom** [comment]
Ahhh, no deprecation warnings.  Feels so nice.

**Comment by dgmstuart** [comment]
@dsandstrom Would it be cleaner to rebase this branch onto master rather than cherrypicking?

**Comment by dsandstrom** [comment]
Yes, I just didn't want to deal with conflicts.

**Comment by dgmstuart** [comment]
@dsandstrom. There aren't any conflicts - I wouldn't have expected there to be any.

I'll force-push the rebased branch now.

**Comment by dgmstuart** [comment]
Fear not the rebase. The rebase is your friend.

**Comment by dgmstuart** [comment]
Rebased again

**Comment by dsandstrom** [comment]
A man with no fears.

**Comment by dgmstuart** [comment]
Ok, I've pushed v2.0.0 as the last foundation-5 version and created the foundation-5 branch there - I think this is ready to become Master. 

@dsandstrom is it ready to be released as 3.0.0, or is there more to do? Maybe better to get that last PR merged first.

**Comment by dsandstrom** [comment]
Since the API is still changing, we shouldn't release 3.0.0, but we should merge this into master.  If we want to release 3.0.0.beta1, it wouldn't hurt.

**Comment by dgmstuart** [comment]
@dsandstrom cool - I wonder if there's value in keeping the foundation-6 branch around?

**Comment by dsandstrom** [comment]
No

**Comment by dsandstrom** [comment]
Well, #152 is a PR on foundation-6.  That should be changed and rebased on master.  It should be a easier rebase.

**Comment by dgmstuart** [comment]
@dsandstrom On second thoughts I'm nuking the foundation-6 branch: it'd be annoying if we got more PRs on it. Master is the branch for Foundation 6 - nice and simeple

**Comment by dgmstuart** [comment]
@dsandstrom high-five - I think everything is where it should be, and we have zero PRs ✋

**Pull request merged**